pkg/auth/tokenauthenticator.go

```diff
@@ -66,3 +79,26 @@ func IsAuthTypePAT(token string) bool {
 	}
 	return false
 }
+
+// ExtractClaimsFromPAT accepts a raw PAT string and returns the PAT claims
+// differs from the imlpementation in oauth.go as Snyk PATs do not strictly follow the JWT spec
+func ExtractClaimsFromPAT(raw string) (*Claims, error) {
+	parts := strings.Split(raw, delimiter)
+	if len(parts) != 4 {
+		return nil, fmt.Errorf("invalid number of segments: %d", len(parts))
+	}
+
+	payload, err := base64.RawURLEncoding.DecodeString(parts[2])
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode payload: %w", err)
+	}
+
+	var c Claims
+	if err = json.Unmarshal(payload, &c); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal payload: %w", err)
+	}
+
+	return &Claims{
```

Comment:
Question: maybe an obsolete question due to [this proposal](https://github.com/snyk/go-application-framework/pull/395/files#r2254143585) but why can't we just return `c`?